src/OICodeGen.cpp

```diff
@@ -1247,6 +1353,41 @@ bool OICodeGen::isEmptyClassOrFunctionType(drgn_type *type,
           drgn_type_num_members(type) == 0);
 }
 
+/*
+ * Returns true iff the provided type
+ */
+bool OICodeGen::isDynamic(drgn_type *type) const {
+  if (!config.polymorphicInheritance) {
```

Comment:
You could group this and the following if together with an || as they both return false in the failure case.